### PR TITLE
[MUSA][Diffusion] Improve  wan model inference speed using torch.compile

### PR DESCRIPTION
**Pull request by wenqf11**
## Motivation

Improve Wan model inference using `torch.compile` on MUSA devicem, also on CUDA.

## Modifications

Add torch.compile for MulAdd, Layernorm with ScaleShift in SGLang Diffusion.

## Speed Tests and Profiling

**Devices: MT S5000**
**MUSA: 4.3.6**
**Model: Wan2.2-T2V-A14B-Diffusers**
**Single step improved: 13.14/11.97=1.09x**

before changes:
s5000-berfore-changes
after changes:
s5000-after-changes

**Devices: H200**
**CUDA: 13.0**
**Model: Wan2.1-T2V-1.3B-Diffusers**
**Single step improved: 0.3381/0.3206=1.05x**
before changes:
h200-before-changes
after changes:
h200-after-changes

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): [Run #25981392374](https://github.com/sgl-project/sglang/actions/runs/25981392374)
Latest PR Test (Extra): :warning: **Not enabled** — add `run-ci-extra` label to opt in.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/models/dits/wanvideo.py [hunk 1]
```diff
@@ -1049,7 +1049,7 @@ def forward(
             )
 
         hidden_states = self.patch_embedding(hidden_states)
-        hidden_states = hidden_states.flatten(2).transpose(1, 2)
+        hidden_states = hidden_states.flatten(2).transpose(1, 2).contiguous()
```
Comment (yeahdongcn) [inline python/sglang/multimodal_gen/runtime/models/dits/wanvideo.py:1053]: Based on our offline discussion, this change should only apply to MUSA. Could you add an `is_musa` check here to limit the behavior to MUSA only? Thanks!

Comment (wenqf11) [reply python/sglang/multimodal_gen/runtime/models/dits/wanvideo.py:1053]: image

add `contiguous()` here has no effect since nn.Module: LayerNormScaleShift_0 will automatic do  `contiguous()` without this modification.

**Review by yeahdongcn — APPROVED** [review]
Looks good from my side. Please @mickqian also take a look. Thanks!

**Review by mickqian — APPROVED** [review]
